libs/sdk-ui-filters/src/AttributeFilter/Components/ElementsSelect/AttributeFilterVirtualisedElementsSelect.tsx

```diff
@@ -0,0 +1,233 @@
+    enableAttributeFilterVirtualised?: boolean;
+}
+
+/**
+ * This component allows users to search Attribute Filter elements.
+ * It manipulates selection.
+ * It displays statues like loading, filtering etc.
+ * It allows paging.
+ * It displays selection status like number of elements and selected elements labels.
+ * It displays messages when elements are filtered out by parent filters or the result of search is empty.
+ *
+ * @internal
+ */
+export const AttributeFilterVirtualisedElementsSelect: React.FC<IAttributeFilterVirtualisedElementsSelect> = (
+    props,
+) => {
+    const {
+        items,
+        totalItemsCount,
+        totalItemsCountWithCurrentSettings,
+
+        isInverted,
+        selectedItems,
+        onSelect,
+
+        searchString,
+        onSearch,
+
+        isLoading,
+        isLoadingNextPage,
+        nextPageSize,
+        onLoadNextPage,
+        error,
+
+        attributeTitle,
+        isFilteredByParentFilters,
+        parentFilterTitles,
+
+        enableShowingFilteredElements,
+        onShowFilteredElements,
+
+        irrelevantSelection,
+        onClearIrrelevantSelection,
+
+        onApplyButtonClick,
+        isApplyDisabled,
+
+        isFilteredByDependentDateFilters,
+        withoutApply,
+    } = props;
+
+    const intl = useIntl();
+    const isMobile = useMediaQuery("mobileDevice");
+    const { fullscreenOnMobile, selectionMode, attribute } = useAttributeFilterContext();
+    const {
+        ElementsSelectLoadingComponent,
+        ElementsSelectItemComponent,
+        ElementsSelectErrorComponent,
+        EmptyResultComponent,
+        ElementsSearchBarComponent,
+        ElementsSelectActionsComponent,
+        StatusBarComponent,
+    } = useAttributeFilterComponentsContext();
+
+    const itemHeight = fullscreenOnMobile && isMobile ? MOBILE_LIST_ITEM_HEIGHT : ITEM_HEIGHT;
+    const isAdaptiveHeight = isMobile && fullscreenOnMobile;
+
+    const previousItemsCount = usePrevious(totalItemsCountWithCurrentSettings);
+    const loadingHeight = useMemo(() => {
+        return Math.max((Math.min(previousItemsCount, VISIBLE_ITEMS_COUNT) || 1) * itemHeight, 20) + 32;
+    }, [previousItemsCount, itemHeight]);
+
+    const primaryLabelTitle = useMemo(() => {
+        if (!attribute) {
+            return null;
+        }
+        const primaryLabel = attribute.displayForms.find((df) => !!df.isPrimary);
+        return primaryLabel?.title;
+    }, [attribute]);
+
+    return (
+        <>
+            <InvertableSelectVirtualised<IAttributeElement>
+                className="gd-attribute-filter-elements-select__next"
+                adaptiveWidth
+                adaptiveHeight={isAdaptiveHeight}
+                isSingleSelect={selectionMode === "single"}
+                items={items}
+                totalItemsCount={totalItemsCountWithCurrentSettings}
+                itemHeight={itemHeight}
+                getItemKey={getElementKey}
+                getItemTitle={(item) => {
```

Comment:
Nitpick, but I would inline this to `getItemTitle={(item) => getElementTitle(item, intl)}`

Reply:
You can possibly inline few others below as well or just remove useless return, i.e.

```
renderItem={(props) => {
    return (
        <ElementsSelectItemComponent
            {...props}
            primaryLabelTitle={primaryLabelTitle}
            fullscreenOnMobile={fullscreenOnMobile}
        />
    );
}}
```
to

```
renderItem={(props) => (
      <ElementsSelectItemComponent
          {...props}
          primaryLabelTitle={primaryLabelTitle}
          fullscreenOnMobile={fullscreenOnMobile}
      />
)}
```